keyvault/client/secrets/README.md

```diff
@@ -0,0 +1,337 @@
+```Java
+SecretAsyncClient secretAsyncClient = SecretAsyncClient.builder()
+        .endpoint(<your-vault-url>)
+        .credential(new AzureCredential())
+        .build();
+
+secretAsyncClient.setSecret("secret_name", "secret_value").subscribe(secretResponse ->
+  System.out.printf("Secret is created with name %s and value %s \n", secretResponse.value().name(), secretResponse.value().value()));
+```
+
+### Retrieve a Secret Asynchronously
+
+Retrieve a previously stored Secret by calling `getSecret`.
+```Java
+SecretAsyncClient secretAsyncClient = SecretAsyncClient.builder()
+        .endpoint(<your-vault-url>)
+        .credential(new AzureCredential())
+        .build();
+        
+secretAsyncClient.getSecret("secretName").subscribe(secretResponse ->
+  System.out.printf("Secret with name %s , value %s \n", secretResponse.value().name(),
+  secretResponse.value().value()));
+```
+
+### Update an existing Secret Asynchronously
+
+Update an existing Secret by calling `updateSecret`.
+```Java
+SecretAsyncClient secretAsyncClient = SecretAsyncClient.builder()
+        .endpoint(<your-vault-url>)
+        .credential(new AzureCredential())
+        .build();
+
+secretAsyncClient.getSecret("secretName").subscribe(secretResponse -> {
+     // Get the Secret
+     Secret secret = secretResponse.value();
+     // Update the expiry time of the secret.
+     secret.expires(OffsetDateTime.now().plusDays(50));
+     secretAsyncClient.updateSecret(secret).subscribe(secretResponse ->
+         System.out.printf("Secret's updated not before time %s \n", secretResponse.value().notBefore().toString()));
+   });
+```
+
+### Delete a Secret Asynchronously
+
+Delete an existing Secret by calling `deleteSecret`.
+```Java
+SecretAsyncClient secretAsyncClient = SecretAsyncClient.builder()
+        .endpoint(<your-vault-url>)
+        .credential(new AzureCredential())
+        .build();
+        
+secretAsyncClient.deleteSecret("secretName").subscribe(deletedSecretResponse ->
+   System.out.printf("Deleted Secret's deletion time %s \n", deletedSecretResponse.value().deletedDate().toString()));
+```
+
+### List Secrets Asynchronously
+
+List the secrets in the key vault by calling `listSecrets`.
+```Java
+SecretAsyncClient secretAsyncClient = SecretAsyncClient.builder()
+        .endpoint(<your-vault-url>)
+        .credential(new AzureCredential())
+        .build();
+       
+// The List Secrets operation returns secrets without their value, so for each secret returned we call `getSecret` to get its // value as well.
+secretAsyncClient.listSecrets()
+  .flatMap(secretAsyncClient::getSecret).subscribe(secretResponse ->
+    System.out.printf("Secret with name %s , value %s \n", secretResponse.value().name(), secretResponse.value().value()));
+```
+
+## Troubleshooting
+### General
+Key Vault clients raise exceptions defined in azure-core. For more detailed infromation about exceptions and how to deal with them, see [Azure Core exceptions][azure_core_exceptions].
```

Comment:
Linking out to the package of exceptions is probably not helpful. In time we will have more documentation that makes this clearer, but I think for now this reference should go.

Reply:
Removed.